From 683f7c9e9a3985ac3885472c2888548069712be2 Mon Sep 17 00:00:00 2001
From: Nguyen Tran <nguyen.tran.pz@bp.renesas.com>
Date: Fri, 22 Aug 2025 10:20:14 +0700
Subject: [PATCH 2/2] arm64: dts: renesas: r8a779g{0,3,4,5}-du-native: Add DU
 Native device support

    - Enable UIO drivers for all but display along with native
      Linux display (DRM/KMS).
    - Enable 4 lane PCIe by default.

Signed-off-by: Nguyen Tran <nguyen.tran.pz@bp.renesas.com>
---
 arch/arm64/boot/dts/renesas/Makefile          |   8 +-
 .../renesas/r8a779g0-whitehawk-du-native.dts  |  19 +++
 .../renesas/r8a779g3-whitehawk-du-native.dts  |  20 +++
 .../renesas/r8a779g4-whitehawk-du-native.dts  |  20 +++
 .../renesas/r8a779g5-whitehawk-du-native.dts  |  20 +++
 .../renesas/whitehawk-du-native-common.dtsi   | 128 ++++++++++++++++++
 6 files changed, 211 insertions(+), 4 deletions(-)
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-du-native.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-du-native.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-du-native.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-du-native.dts
 create mode 100644 arch/arm64/boot/dts/renesas/whitehawk-du-native-common.dtsi

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index 38a93a0a39d26..ad3eb7bab3901 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -74,10 +74,10 @@ dtb-$(CONFIG_ARCH_R8A779A0) += r8a779a0-falcon.dtb
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g0-whitehawk.dtb r8a779g0-whitehawk-rgid.dtb
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g0-whitehawk-sound.dtb r8a779g0-whitehawk-sound-rgid.dtb
 
-dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g0-whitehawk-native.dtb
-dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-whitehawk-native.dtb
-dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g4-whitehawk-native.dtb
-dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g5-whitehawk-native.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g0-whitehawk-native.dtb r8a779g0-whitehawk-du-native.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-whitehawk-native.dtb r8a779g3-whitehawk-du-native.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g4-whitehawk-native.dtb r8a779g4-whitehawk-du-native.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g5-whitehawk-native.dtb r8a779g5-whitehawk-du-native.dtb
 
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-white-hawk.dtb r8a779g3-white-hawk-rgid.dtb
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-whitehawk-sound.dtb r8a779g3-whitehawk-sound-rgid.dtb
diff --git a/arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-du-native.dts b/arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-du-native.dts
new file mode 100644
index 0000000000000..fe98eaf75cc67
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-du-native.dts
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H White Hawk board
+ * Support Native Linux display driver and 4 lane PCIe.
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r8a779g0.dtsi"
+#include "white-hawk-cpu-common.dtsi"
+#include "white-hawk-common.dtsi"
+#include "white-hawk-native-common.dtsi"
+#include "whitehawk-du-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g0";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-du-native.dts b/arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-du-native.dts
new file mode 100644
index 0000000000000..3e803849080c7
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-du-native.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H-7 White Hawk board
+ * Support Native Linux display driver and 4 lane PCIe.
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r8a779g3.dtsi"
+#include "white-hawk-cpu-common.dtsi"
+#include "white-hawk-common.dtsi"
+#include "white-hawk-native-common.dtsi"
+#include "whitehawk-du-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g3";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0",
+				"renesas,r8a779g3";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-du-native.dts b/arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-du-native.dts
new file mode 100644
index 0000000000000..dd93491a83878
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-du-native.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H-5 White Hawk board
+ * Support Native Linux display driver and 4 lane PCIe.
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r8a779g4.dtsi"
+#include "white-hawk-cpu-common.dtsi"
+#include "white-hawk-common.dtsi"
+#include "white-hawk-native-common.dtsi"
+#include "whitehawk-du-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g4";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0",
+				"renesas,r8a779g4";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-du-native.dts b/arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-du-native.dts
new file mode 100644
index 0000000000000..e1cb25f544b6c
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-du-native.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H-3 White Hawk board
+ * Support Native Linux display driver and 4 lane PCIe.
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r8a779g5.dtsi"
+#include "white-hawk-cpu-common.dtsi"
+#include "white-hawk-common.dtsi"
+#include "white-hawk-native-common.dtsi"
+#include "whitehawk-du-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g5";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0",
+				"renesas,r8a779g5";
+};
diff --git a/arch/arm64/boot/dts/renesas/whitehawk-du-native-common.dtsi b/arch/arm64/boot/dts/renesas/whitehawk-du-native-common.dtsi
new file mode 100644
index 0000000000000..996f9e526d0f8
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/whitehawk-du-native-common.dtsi
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Common Device Tree Source support Native Linux display driver
+ * and 4 lane PCIe for the White Hawk CPU and Breakout boards
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+/ {
+	soc: soc {
+		du0 {
+			status = "disabled";
+		};
+
+		du_01 {
+			status = "disabled";
+		};
+
+		dsi0 {
+			status = "disabled";
+		};
+
+		dsi0_01 {
+			status = "disabled";
+		};
+
+		dsi0_02 {
+			status = "disabled";
+		};
+
+		dsi0_03 {
+			status = "disabled";
+		};
+
+		dsi0_04 {
+			status = "disabled";
+		};
+
+		dsi0_05 {
+			status = "disabled";
+		};
+
+		dsi0_06 {
+			status = "disabled";
+		};
+
+		dsi0_07 {
+			status = "disabled";
+		};
+
+		dsi0_08 {
+			status = "disabled";
+		};
+
+		dsi1 {
+			status = "disabled";
+		};
+
+		dsi1_01 {
+			status = "disabled";
+		};
+
+		dsi1_02 {
+			status = "disabled";
+		};
+
+		dsi1_03 {
+			status = "disabled";
+		};
+
+		dsi1_04 {
+			status = "disabled";
+		};
+
+		dsi1_05 {
+			status = "disabled";
+		};
+
+		dsi1_06 {
+			status = "disabled";
+		};
+
+		dsi1_07 {
+			status = "disabled";
+		};
+
+		dsi1_08 {
+			status = "disabled";
+		};
+
+		doc0: doc@feba0000 {
+			status = "disabled";
+		};
+
+		doc0_01 {
+			status = "disabled";
+		};
+
+		dsc0: dsc@feb8d000 {
+			status = "disabled";
+		};
+	};
+};
+
+&i2c1 {
+	gmsl0: gmsl-deserializer@49 {
+		status = "disabled";
+	};
+
+	gmsl1: gmsl-deserializer@4b {
+		status = "disabled";
+	};
+};
+
+&pciec0 {
+	reset-gpio = <&io_expander_a 0 GPIO_ACTIVE_LOW>;
+	num-lanes = <4>;
+	clocks = <&cpg CPG_MOD 624>, <&cpg CPG_MOD 625>;
+	clock-names = "pcie0_clk", "pcie1_clk";
+	status = "okay";
+};
+
+&pciec0_ep {
+	num-lanes = <4>;
+	clocks = <&cpg CPG_MOD 624>, <&cpg CPG_MOD 625>;
+	clock-names = "pcie0_clk", "pcie1_clk";
+	status = "disabled";
+};
-- 
2.25.1

